docs/source/extension/extension_multiple_ui.rst

```diff
@@ -0,0 +1,277 @@
+.. Copyright (c) Jupyter Development Team.
+.. Distributed under the terms of the Modified BSD License.
+
+.. _multiple_ui_extensions:
+
+Extensions Targeting Multiple Applications
+==========================================
+
+*Make your extensions work in JupyterLab 4, Notebook 7 and more!*
+
+Jupyter Notebook 7 is built with components from JupyterLab, and since
+both use the same building blocks, that means your extension can work
+on both (or any other frontend built with JupyterLab components) with
+little or no modification depending on its design.
+
+This guide will give you an overview of compatibility features, then a
+tutorial and reference code covering some of the topics mentioned here.
+If you don't know how to make extensions, you can read more about the
+basics at :ref:`the Extension Tutorial <extension_tutorial>` or the
+:ref:`the Extensions page <extension_dev>`.
+
+How Compatibility Works
+-----------------------
+
+At a high level, extensions for JupyterLab and Jupyter Notebook both
+typically start from a template project, which you can download and set up
+using instructions from `the extension tutorial <https://jupyterlab.readthedocs.io/en/latest/extension/extension_tutorial.html>`_).
+Once your template is ready, you can start adding components and features to build your extension.
+
+An extension for JupyterLab (and for Notebook 7) is made up of a `series <https://jupyterlab.readthedocs.io/en/latest/extension/extension_dev.html>`_
+of bundled `plugins <https://lumino.readthedocs.io/en/latest/api/interfaces/application.IPlugin.html#requires>`_,
+and those plugins typically use components from the interface toolkit `Lumino <https://lumino.readthedocs.io/en/latest/api/index.html>`_
+as well as the `JupyterLab API <https://jupyterlab.readthedocs.io/en/latest/api/index.html>`_
+(among others) to help build your extension's look and behavior (both are
+written in Typescript).
+
+This is how basic compatibility features work: both apps use the same building
+blocks and methods. For instance, both JupyterLab and Notebook 7 accept Lumino widgets
+as interface components, both apps allow you to add them to the interface by
+specifying an "area" to place them into, and extensions for both use the same
+basic ``JupyterFrontendPlugin`` class.
+
+How to Achieve Compatibility
+----------------------------
+
+Compatibility can be achieved with literally no effort in some simple
```

Comment:
```suggestion
Compatibility can be achieved without additional effort in some simple
```

or with "little effort". "literally no effort" will be frustrating when it does not work for the user (see https://jupyterlab.readthedocs.io/en/latest/developer/contributing.html#writing-style)